src/stan/services/experimental/advi/fullrank.hpp

```diff
@@ -0,0 +1,112 @@
+#ifndef STAN_SERVICES_EXPERIMENTAL_ADVI_FULLRANK_HPP
+#define STAN_SERVICES_EXPERIMENTAL_ADVI_FULLRANK_HPP
+
+#include <stan/callbacks/interrupt.hpp>
+#include <stan/callbacks/writer.hpp>
+#include <stan/services/util/experimental_message.hpp>
+#include <stan/services/util/initialize.hpp>
+#include <stan/services/util/rng.hpp>
+#include <stan/io/var_context.hpp>
+#include <stan/variational/advi.hpp>
+#include <boost/random/additive_combine.hpp>
+#include <string>
+#include <vector>
+
+namespace stan {
+  namespace services {
+    namespace experimental {
+      namespace advi {
+
+        /**
+         * Runs fullrank ADVI.
+         *
+         * @tparam Model A model implementation
+         *
+         * @param model Input model to test (with data already instantiated)
+         * @param init var context for initialization
+         * @param random_seed random seed for the pseudo random number generator
+         * @param chain chain id to advance the pseudo random number generator
+         * @param init_radius radius to initialize
+         * @param grad_samples number of samples for Monte Carlo estimate of gradients
+         * @param elbo_samples number of samples for Monte Carlo estimate of ELBO
+         * @param max_iterations maximum number of iterations
+         * @param tol_rel_obj convergence tolerance on the relative norm of the objective
+         * @param eta stepsize scaling parameter for variational inference
+         * @param adapt_engaged adaptation engaged?
+         * @param adapt_iterations number of iterations for eta adaptation
+         * @param eval_elbo evaluate ELBO every Nth iteration
+         * @param output_samples number of posterior samples to draw and save
+         * @param[out] interrupt interrupt callback to be called every iteration
+         * @param[out] message_writer output for messages
+         * @param[out] init_writer Writer callback for unconstrained inits
+         * @param[out] parameter_writer output for parameter values
+         * @param[out] diagnostic_writer output for diagnostic values
+         * @return stan::services::error_codes::OK if successful
+         */
+        template <class Model>
```

Comment:
Yes, I believe so. I don't think `prob_grad` has enough of the methods -- we've just tacked on additional methods through the generator instead of actually updating the appropriate base class. So yes, let's create an issue for that.